OfficeDev/microsoft-teams-library-js
# Add React Router for Tab Navigation in the Test App

For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

This PR adds React Router to enable navigation between tabs when caching is enabled. There is no Nav component because tabs are populated in the appDefinition files. The onResume function is changed to navigate based on the URL provided. 

## Validation

### Validation performed:

1. change the URL to https://localhost:4000/second-page
2. ensure the content is an empty page
3. switch tabs with caching enabled in a test host and ensure the onResume function properly navigates to the empty page.

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

<Yes/No>

### End-to-end tests added:

<Yes/No>

## Additional Requirements

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

<Yes/No>

### Related PRs:

> Remove this section if n/a

### Next/remaining steps:

> List the next or remaining steps in implementing the overall feature in subsequent PRs (or is the feature 100% complete after this?).

> Remove this section if n/a

- [ ] Item 1
- [ ] Item 2

### Screenshots:

> Remove this section if n/a

| Before     | After      |
| ---------- | ---------- |
| < image1 > | < image2 > |
